platform/ext/target/nordic_nrf/common/nrf5340/tfm_hal_its_encryption.c

```diff
@@ -0,0 +1,196 @@
+#include "platform/include/tfm_platform_system.h"
+#include "platform/include/tfm_hal_its.h"
+#include "psa/crypto.h"
+
+#include "nrf_cc3xx_platform_derived_key.h"
+#include "nrf_cc3xx_platform.h"
+#include "nrf_cc3xx_platform_kmu.h"
+#include <string.h>
+
+#define ITS_ENCRYPTION_SUCCESS 0
+
+#define HUK_KMU_SLOT 2
+#define HUK_KMU_SIZE_BITS 256
+
+static bool ctx_is_valid(struct tfm_hal_its_aead_ctx *ctx)
+{
+    bool ret;
+
+    if (ctx == NULL) {
+        return false;
+    }
+
+    ret = (ctx->deriv_label == NULL && ctx->deriv_label_size != 0) ||
+          (ctx->aad == NULL && ctx->aad_size != 0) ||
+          (ctx->nonce == NULL && ctx->nonce_size != 0) ||
+          (ctx->tag == NULL && ctx->tag_size != 0);
+
+    return !ret;
+}
+
+static int32_t tfm_hal_its_aead_crypt(uint8_t *plaintext,
```

Comment:
Propose to convert this to a static init-function and move encrypt/decrypt out of this file
Also propose to make a static free-function to qualify clearing out the context at the end of the function

Reply:
Should be done now